atc/worker/image_factory.go

```diff
@@ -59,3 +61,6 @@ type NoopImageFetchingDelegate struct{}
 func (NoopImageFetchingDelegate) Stdout() io.Writer                                 { return ioutil.Discard }
 func (NoopImageFetchingDelegate) Stderr() io.Writer                                 { return ioutil.Discard }
 func (NoopImageFetchingDelegate) ImageVersionDetermined(db.UsedResourceCache) error { return nil }
+func (NoopImageFetchingDelegate) ImageSourceRedaction(source atc.Source) (atc.Source, error) {
+	return source, nil
```

Comment:
I'm not sure if it's OK for this to be a no-op - it's used during the `check` for an `image_resource:` image, so we may need credentials to be redacted in that case too.

Reply:
yeah, good catch! As it's noop, we may just return an empty source as redacted.